src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -3,12 +3,63 @@
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
+import core.basesyntax.utility.RegistrationException;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final Integer MIN_AGE = 18;
+    private static final Integer MIN_SYMBOLS = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User should not be null");
+        }
+        validateUserLogin(user);
+        validateUserAge(user);
+        validateUserPassword(user);
+        if (storageDao.get(user.getLogin()) == null) {
+            storageDao.add(user);
+        }
+        return user;
+    }
+
+    private void validateUserPassword(User user) {
+        if (user.getPassword().length() < MIN_SYMBOLS) {
+            throw new RegistrationException("User password should not be less than "
+                    + MIN_SYMBOLS + " symbols");
+        }
+        if (user.getPassword().isEmpty()) {
+            throw new RegistrationException("User password should not be empty");
+        }
+        if (user.getPassword() == null) {
```

Comment:
think about the order of those checks. What will happen if you try to call  `.isEmpty` , `.length()` on null?

The null check for the user password should be performed before checking its length or if it's empty. Move this check to be the first in the 'validateUserPassword' method.

Reply:
Done